hudi-common/src/main/java/org/apache/hudi/common/config/SerializableSchema.java

```diff
@@ -62,12 +62,17 @@ private void readObject(ObjectInputStream in) throws IOException {
   
   // create a public write method for unit test
   public void writeObjectTo(ObjectOutputStream out) throws IOException {
-    out.writeUTF(schema.toString());
+    // Note: writeUTF cannot support string length > 64K. So use writeObject which has small overhead (relatively).
+    out.writeObject(schema.toString());
```

Comment:
Instead of this, can we do : 
byte[] data = schema.toString.getBytes("UTF-8");
out.writeBytes(data) to ensure we don't lost UTF encoding ?

Reply:
This was initial implementation. Unfortunately, it doesnt well with java serialization because we have 'out.defaultWriteObject();' in front.  Per Vinoth's suggestion I used writeUTF, see https://issues.apache.org/jira/browse/HUDI-1443 for some context.